apps/dot_ext/utils.py

```diff
@@ -34,3 +41,25 @@ def remove_application_user_pair_tokens_data_access(application, user):
         refresh_token_delete_cnt = RefreshToken.objects.filter(application=application, user=user).delete()[0]
 
     return data_access_grant_delete_cnt, access_token_delete_cnt, refresh_token_delete_cnt
+
+
+def validate_app_is_active(request):
+
+    client_id = None
+
+    if request.GET is not None:
+        client_id = request.GET.get('client_id', None)
+
+    if client_id is None and request.POST is not None:
+        client_id = request.POST.get('client_id', None)
```

Comment:
I think it would be a bit cleaner to use `elif` here:
```suggestion
    if request.GET is not None:
        client_id = request.GET.get('client_id', None)
    elif request.POST is not None:
        client_id = request.POST.get('client_id', None)
```

Reply:
will try this out and get back

Reply:
it seems the request object is instance of WSGIRequest and always has GET and POST which are <QueryDict>, when request method is POST, then POST has entries and GET is an empty dict, when request method is GET, GET has entries and POST is an empty dict, on the safe side, I still check if request.POST / GET is not None, and change the code to if elif...
![image](https://user-images.githubusercontent.com/25412191/95248208-d3e9c300-07cb-11eb-8e18-f6ec55024d19.png)
`def validate_app_is_active(request):

    client_id = None

    if request.GET is not None and request.GET.get('client_id', None) is not None:
        client_id = request.GET.get('client_id', None)
    elif request.POST is not None and request.POST.get('client_id', None):
        client_id = request.POST.get('client_id', None)

    app = None

    if client_id is not None:
        try:
            app = Application.objects.get(client_id=client_id)
        except Application.DoesNotExist:
            pass

    if app and not app.active:
        raise PermissionDenied(settings.APPLICATION_TEMPORARILY_INACTIVE.format(app.name))
`